src/renderers/shared/fiber/ReactFiberBeginWork.js

```diff
@@ -436,6 +438,16 @@ module.exports = function<T, P, I, TI, C, CX>(
     } else {
       // Proceed under the assumption that this is a functional component
       workInProgress.tag = FunctionalComponent;
+      if (__DEV__) {
+        warning(
+          !!workInProgress.ref,
+          'Stateless function components cannot be given refs ' +
+          '(See ref "' + workInProgress.ref._stringRef + '" in ' +
+          getComponentName(workInProgress) + ' created by ' +
+          getComponentName(workInProgress.return) + '). ' +
```

Comment:
Let's change https://github.com/facebook/react/blob/master/src/renderers/shared/shared/__tests__/ReactStatelessComponent-test.js#L149 to put component in a `div` to make sure we're printing owner rather than parent name.